### ISSUE TITLE
Upgrade logos dependency in lunarity

### ISSUE THREAD

**Issue by birchmd** [aurora-is-near/aurora-engine#34]
Our project has a dependency that does not compile on latest nightly

```
$ cargo +nightly check
   Compiling logos-derive v0.7.7
error[E0061]: this function takes 1 argument but 2 arguments were supplied
   --> /home/birchmd/.cargo/registry/src/github.com-1ecc6299db9ec823/logos-derive-0.7.7/src/lib.rs:55:20
    |
55  |             extras.insert(util::ident(&ext), |_| panic!("Only one #[extras] attribute can be declared."));
    |                    ^^^^^^ -----------------  ----------------------------------------------------------- supplied 2 arguments
    |                    |
    |                    expected 1 argument
    |
note: associated function defined here

error[E0061]: this function takes 1 argument but 2 arguments were supplied
   --> /home/birchmd/.cargo/registry/src/github.com-1ecc6299db9ec823/logos-derive-0.7.7/src/lib.rs:89:23
    |
89  |                 error.insert(variant, |_| panic!("Only one #[error] variant can be declared."));
    |                       ^^^^^^ -------  -------------------------------------------------------- supplied 2 arguments
    |                       |
    |                       expected 1 argument
    |
note: associated function defined here

error[E0061]: this function takes 1 argument but 2 arguments were supplied
   --> /home/birchmd/.cargo/registry/src/github.com-1ecc6299db9ec823/logos-derive-0.7.7/src/lib.rs:93:21
    |
93  |                 end.insert(variant, |_| panic!("Only one #[end] variant can be declared."));
    |                     ^^^^^^ -------  ------------------------------------------------------ supplied 2 arguments
    |                     |
    |                     expected 1 argument
    |
note: associated function defined here

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0061`.
error: could not compile `logos-derive`

To learn more, run the command again with --verbose.
```

This `logos-derive` dependency comes from our [dependency on `lunarity`](https://github.com/aurora-is-near/aurora-engine/blob/8b767e44a5378da090c3ceb9113af4d7378f9216/Cargo.toml#L49). Fortunately, we are already working off a fork which we control (it's on Illia's GitHub), and the latest version of `logos` (v0.12) _does_ compile with latest nightly. So we can resolve this issue by upgrading `logos` to the latest version in our fork of the `lunarity` project.

Unfortunately, this upgrade is non-trivial because there were [several breaking API changes](https://github.com/maciejhirsz/logos/releases/tag/v0.11.0) in `logos`, so some code in `lunarity` will need to be refactored to work with the latest version. This issue represents the work to do that refactoring and thus have a version of `lunarity` that compiles in latest nightly for our project to depend on.

This is probably a low priority item since it will likely not be blocking us until we decide to upgrade our `rust-toolchain` file to a later version of rust.

**Comment by birchmd** [comment]
Also see https://github.com/maciejhirsz/logos/issues/105 for more information on the breaking API changes.

**Comment by joshuajbouw** [comment]
It seems like this has been fixed by the `logos` team and now in `0.11.0-rc1`.

**Comment by birchmd** [comment]
> It seems like this has been fixed by the logos team and now in 0.11.0-rc1.

Yes, I know, that's what this issue is about. Recent `logos` version do compile under latest nightly, but we need to update `lunarity` (the direct dependency of ours which depends on `logos`) so that it uses a more recent version. However, this is non-trivial because the API changes in `logos`.

**Comment by birchmd** [comment]
Fixed in #115
